plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/BaseJdbcClient.java

```diff
@@ -305,7 +305,16 @@ public List<JdbcColumnHandle> getColumns(ConnectorSession session, JdbcTableHand
                         getInteger(resultSet, "DECIMAL_DIGITS"),
                         Optional.empty(),
                         Optional.empty());
-                Optional<ColumnMapping> columnMapping = toColumnMapping(session, connection, typeHandle);
+                Optional<ColumnMapping> columnMapping;
+                try {
+                    columnMapping = toColumnMapping(session, connection, typeHandle);
+                }
+                catch (RuntimeException e) {
```

Comment:
This sounds too wide. For example NPE in Trino code should manifest loudly. Can we capture only `SqlException`?

Reply:
No. There are cases independent from trino: for some connectors jdbc driver could be broken or updated to a newer but broken version. E.g. for some type jdbc driver could return negative values for type, length, precision, scale. In such situation `toColumnMapping` will silently becomes broken, which makes not possible metadata queries on columns. E.g. to list all table columns.

For example postgres will fail in such case (unfortunately there are no tests for this):
```
    @Override
    public Optional<ColumnMapping> toColumnMapping(ConnectorSession session, Connection connection, JdbcTypeHandle typeHandle)
    {
        String jdbcTypeName = typeHandle.getJdbcTypeName()
                .orElseThrow(() -> new TrinoException(JDBC_ERROR, "Type name is missing: " + typeHandle)); 
```

Hmm.. But may be it's enough to provide more user friendly exception message instead of ignoring this column.

Reply:
https://github.com/trinodb/trino/pull/15056